app/src/main/java/com/duckduckgo/app/browser/BrowserTrackersAnimatorHelper.kt

```diff
@@ -171,42 +98,45 @@ class BrowserTrackersAnimatorHelper @Inject constructor() : CoroutineScope {
         }
     }
 
-    private fun createCompleteTrackersAnimation(
+    private fun createFullTrackersAnimation(
         activity: Activity,
         container: ConstraintLayout,
-        loadingText: View,
         views: List<View>,
         events: List<TrackingEvent>?
     ): AnimatorSet {
         return AnimatorSet().apply {
-            play(createTrackersAnimation(activity, container, loadingText, events))
+            play(createPartialTrackersAnimation(activity, container, views, events))
             play(animateFadeOut(container))
                 .after(logosStayOnScreenDuration)
                 .before(animateOmnibarIn(views))
         }
     }
 
-    private fun createTrackersAnimation(
+    private fun createPartialTrackersAnimation(
         activity: Activity,
         container: ConstraintLayout,
-        loadingText: View,
+        views: List<View>,
         events: List<TrackingEvent>?
     ): AnimatorSet {
         container.removeAllViews()
         container.alpha = 0f
 
         val logos = createResourcesIdList(activity, events)
-        val views: List<View> = createLogosViewList(activity, container, logos)
-
-        applyConstraintSet(container, views)
-
-        val fadeLoadingOut = animateFadeOut(loadingText)
+        val logoViews: List<View> = createLogosViewList(activity, container, logos)
+        val fadeOmnibarOut = animateOmnibarOut(views)
         val fadeLogosIn = animateFadeIn(container)
 
-        animateBlockedLogos(views)
+        logosStayOnScreenDuration = if (logoViews.isEmpty()) {
```

Comment:
NIT: Is there any way to move this `logosStayOnScreenDuration` as part of `createFullTrackersAnimation` as a `val` and remove the class field?

Reply:
There is and I just done it, this actually made me realised about a small bug. In sites where we have trackers but not logos (i.e. twitch.tv) there was a small glitch with the animation.